Serializing ZoneId throws error

### Expected Behavior

It looks like Serde isn't compatible with ZoneId, it's throwing errors.

### Actual Behaviour

It should work.

### Steps To Reproduce

Serializing my model DTO produces the following error:
```kotlin
@Serdeable
data class Location(
    val timeZoneId: ZoneId,
)
```
```
Caused by: io.micronaut.serde.exceptions.SerdeException: No serializable introspection present for type ZoneRegion. Consider adding Serdeable. Serializable annotate to type ZoneRegion. Alternatively if you are not in control of the project's source code, you can use @SerdeImport(ZoneRegion.class) to enable serialization of this type.
```
After this, I tried to use `@SerdeImport` to import the affected classes. `ZoneRegion` wasn't available for import, maybe because it's internal, but after adding the following imports, I got farther, but I eventually hit a new error:
```kotlin
@SerdeImport(ZoneId::class)
@SerdeImport(ZoneRules::class)
@SerdeImport(ZoneOffsetTransition::class)
```
Final error:
```
18:14:30.726 [default-nioEventLoopGroup-1-3] ERROR i.m.http.server.RouteExecutor - Unexpected error occurred: class com.cfa.ots.fsd.$java_time_zone_ZoneOffsetTransition$Introspection tried to access method 'java.util.List java.time.zone.ZoneOffsetTransition.getValidOffsets()' (com.cfa.ots.fsd.$java_time_zone_ZoneOffsetTransition$Introspection is in unnamed module of loader 'app'; java.time.zone.ZoneOffsetTransition is in module java.base of loader 'bootstrap')
java.lang.IllegalAccessError: class com.cfa.ots.fsd.$java_time_zone_ZoneOffsetTransition$Introspection tried to access method 'java.util.List java.time.zone.ZoneOffsetTransition.getValidOffsets()' (com.cfa.ots.fsd.$java_time_zone_ZoneOffsetTransition$Introspection is in unnamed module of loader 'app'; java.time.zone.ZoneOffsetTransition is in module java.base of loader 'bootstrap')
	at com.cfa.ots.fsd.$java_time_zone_ZoneOffsetTransition$Introspection.dispatchOne(Unknown Source)
	at io.micronaut.inject.beans.AbstractInitializableBeanIntrospection$BeanReadPropertyImpl.getUnsafe(AbstractInitializableBeanIntrospection.java:1170)
	at io.micronaut.serde.support.serializers.SerBean$PropSerProperty.get(SerBean.java:511)
	at io.micronaut.serde.support.serializers.CustomizedObjectSerializer.serializeInto(CustomizedObjectSerializer.java:63)
```

Is there anything else I can try here? I'm considering writing custom serializers, but I wanted to check if there's a better way because this is surely a common problem.

### Environment Information

OSX Sequioa 15.2
JDK 17.0.10

### Example Application

_No response_

### Version

4.7.4

--- a/serde-support/src/main/java/io/micronaut/serde/support/serdes/Serdes.java
+++ b/serde-support/src/main/java/io/micronaut/serde/support/serdes/Serdes.java
@@ -131,6 +131,7 @@ public final class Serdes {
         if (jacksonJsonNodeSerde != null) {
             consumer.accept(jacksonJsonNodeSerde);
         }
+        consumer.accept(new ZoneIdSerde());
     }
 
 }

--- a/serde-support/src/main/java/io/micronaut/serde/support/serdes/ZoneIdSerde.java
+++ b/serde-support/src/main/java/io/micronaut/serde/support/serdes/ZoneIdSerde.java
@@ -1,0 +1,56 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.serde.support.serdes;
+
+import java.io.IOException;
+import java.time.ZoneId;
+
+import io.micronaut.core.type.Argument;
+import io.micronaut.serde.Decoder;
+import io.micronaut.serde.Encoder;
+import io.micronaut.serde.support.SerdeRegistrar;
+
+/**
+ * ZoneId serde.
+ */
+public class ZoneIdSerde implements SerdeRegistrar<ZoneId> {
+    @Override
+    public void serialize(Encoder encoder, EncoderContext context, Argument<? extends ZoneId> type, ZoneId value) throws IOException {
+        if (value != null) {
+            encoder.encodeString(value.getId());
+        } else {
+            encoder.encodeNull();
+        }
+    }
+
+    @Override
+    public ZoneId deserialize(Decoder decoder, DecoderContext context, Argument<? super ZoneId> type) throws IOException {
+        return ZoneId.of(decoder.decodeString());
+    }
+
+    @Override
+    public ZoneId deserializeNullable(Decoder decoder, DecoderContext context, Argument<? super ZoneId> type) throws IOException {
+        if (decoder.decodeNull()) {
+            return null;
+        }
+        return deserialize(decoder, context, type);
+    }
+
+    @Override
+    public Argument<ZoneId> getType() {
+        return Argument.of(ZoneId.class);
+    }
+}

--- a/serde-support/src/main/java/io/micronaut/serde/support/serdes/ZoneIdSerde.java
+++ b/serde-support/src/main/java/io/micronaut/serde/support/serdes/ZoneIdSerde.java
@@ -18,6 +18,7 @@ package io.micronaut.serde.support.serdes;
 import java.io.IOException;
 import java.time.ZoneId;
 
+import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.type.Argument;
 import io.micronaut.serde.Decoder;
 import io.micronaut.serde.Encoder;
@@ -25,15 +26,14 @@ import io.micronaut.serde.support.SerdeRegistrar;
 
 /**
  * ZoneId serde.
+ *
+ * @since 2.16
  */
-public class ZoneIdSerde implements SerdeRegistrar<ZoneId> {
+@Internal
+final class ZoneIdSerde implements SerdeRegistrar<ZoneId> {
     @Override
     public void serialize(Encoder encoder, EncoderContext context, Argument<? extends ZoneId> type, ZoneId value) throws IOException {
-        if (value != null) {
-            encoder.encodeString(value.getId());
-        } else {
-            encoder.encodeNull();
-        }
+        encoder.encodeString(value.getId());
     }
 
     @Override

--- a/serde-support/src/test/groovy/io/micronaut/serde/support/serdes/ZoneIdSerdeSpec.groovy
+++ b/serde-support/src/test/groovy/io/micronaut/serde/support/serdes/ZoneIdSerdeSpec.groovy
@@ -1,0 +1,56 @@
+package io.micronaut.serde.support.serdes
+
+import java.time.ZoneId
+
+import io.micronaut.serde.ObjectMapper
+import io.micronaut.serde.annotation.Serdeable
+import io.micronaut.test.extensions.spock.annotation.MicronautTest
+import jakarta.inject.Inject
+import spock.lang.Specification
+
+@MicronautTest(startApplication = false)
+class ZoneIdSerdeSpec extends Specification {
+    @Inject
+    ObjectMapper objectMapper
+
+    void "ZoneId serialized"(String value) {
+        given:
+        ZoneIdPojo pojo = new ZoneIdPojo(zoneId: ZoneId.of(value))
+
+        when:
+        String json = objectMapper.writeValueAsString(pojo)
+
+        then:
+        json == "{\"zoneId\":\"${value}\"}"
+
+        where:
+        value << ["UTC", "GMT", "Z", "Europe/Moscow", "America/New_York", "UTC+01:00", "GMT-08:00", "UT+05:30", "+11:00", "-10:00"]
+    }
+
+    void "ZoneId deserialized"(String value) {
+        given:
+        String json = "{\"zoneId\":\"${value}\"}"
+
+        when:
+        ZoneIdPojo pojo = objectMapper.readValue(json, ZoneIdPojo)
+
+        then:
+        pojo.getZoneId() == ZoneId.of(value)
+
+        where:
+        value << ["UTC", "GMT", "Z", "Europe/Moscow", "America/New_York", "UTC+01:00", "GMT-08:00", "UT+05:30", "+06", "-03", "+11:00", "-10:00"]
+    }
+
+    @Serdeable
+    static class ZoneIdPojo {
+        private ZoneId zoneId
+
+        ZoneId getZoneId() {
+            this.zoneId
+        }
+
+        void setZoneId(ZoneId zoneId) {
+            this.zoneId = zoneId
+        }
+    }
+}

